### ISSUE TITLE
Switching back to pulseaudio on debian bookworm removes `qubes-vm-recommended`

### ISSUE THREAD

**Issue by Minimalist73** [QubesOS/qubes-issues#8642]
### Qubes OS release
r4.2-rc4

### Brief summary
After having a few issues with pipewire and my microphone, I decided to use pulseaudio again since it's the only working solution for me right now. Switching on fedora 38 was easy but it can't be done on Debian bookworm because of dependencies.

### Steps to reproduce
- Start a terminal in a debian-12 template
- Install `pulseaudio-qubes`

### Expected behavior
Package `pulseaudio-qubes` can be installed to replace `pipewire-qubes`

### Actual behavior
When installing package `pulseaudio-qubes`, it wants to remove both `pipewire-qubes` and `qubes-vm-recommended`.
```
user@test:~$ sudo apt install pulseaudio-qubes
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages were automatically installed and are no longer required:
  curl debugedit deltarpm distro-info-data dnf dnf-data fakeroot fwupd-qubes-vm jcat libcomps0 libdnf2 libdnf2-common libfakeroot libfsverity0 libimagequant0 libjcat1 liblua5.3-0
  libmodulemd2 libpipewire-0.3-modules libraqm0 librepo0 librpm9 librpmbuild9 librpmio9 librpmsign9 libsolv1 libsolvext1 libunbound8 libwireplumber-0.4-0 python-apt-common
  python-babel-localedata python3-apt python3-babel python3-certifi python3-chardet python3-charset-normalizer python3-click python3-colorama python3-croniter python3-dateutil
  python3-distutils python3-dnf python3-gpg python3-hawkey python3-jinja2 python3-jmespath python3-lib2to3 python3-libcomps python3-libdnf python3-looseversion python3-markupsafe
  python3-msgpack python3-numpy python3-olefile python3-packaging python3-pil python3-psutil python3-py python3-pycryptodome python3-qubesimgconverter python3-requests python3-rpm
  python3-tqdm python3-tz python3-unbound python3-yaml python3-zmq qubes-core-agent-dom0-updates qubes-core-agent-passwordless-root qubes-gpg-split qubes-img-converter
  qubes-input-proxy-sender qubes-mgmt-salt-vm-connector qubes-pdf-converter qubes-repo-templates qubes-usb-proxy rpm rpm-common rpm2cpio salt-common salt-ssh sqlite3 usbutils
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  pipewire-qubes qubes-vm-recommended
The following NEW packages will be installed:
  pulseaudio-qubes
0 upgraded, 1 newly installed, 2 to remove and 0 not upgraded.
Need to get 21.1 kB of archives.
After this operation, 14.3 kB disk space will be freed.
Do you want to continue? [Y/n]
```

**Comment by bi0shacker001** [comment]
qubes-vm-recommended has pipewire-qubes as a dependency, since it's the new recommendation.
As pipewire is the new standard qubes uses, I believe this is behaving as intended, as you aren't using the recommended packageset if you use pulse now.

**Comment by unman** [comment]
This is right, and this is not a bug.
If a user chooses to remove a package that is included in
qubes-vm-recommended, (or any other meta package), then they must
protect the other packages that they wish to keep.
This is standard packaging usage.

Issue should be closed.

**Comment by Minimalist73** [comment]
@unman Sorry but I don't agree on this. On Fedora, it is possible to switch back to pulseaudio with a simple `dnf swap` command without any packages being deleted. On Debian, it switch a bunch of packages to "no longer required" which will be completely removed if the user use `apt autoremove`. I don't see why it's normal that the user have to protect all of the following package manually only because of 1 package being deleted:

```
curl debugedit deltarpm distro-info-data dnf dnf-data fakeroot fwupd-qubes-vm jcat libcomps0 libdnf2 libdnf2-common libfakeroot libfsverity0 libimagequant0 libjcat1 liblua5.3-0
  libmodulemd2 libpipewire-0.3-modules libraqm0 librepo0 librpm9 librpmbuild9 librpmio9 librpmsign9 libsolv1 libsolvext1 libunbound8 libwireplumber-0.4-0 python-apt-common
  python-babel-localedata python3-apt python3-babel python3-certifi python3-chardet python3-charset-normalizer python3-click python3-colorama python3-croniter python3-dateutil
  python3-distutils python3-dnf python3-gpg python3-hawkey python3-jinja2 python3-jmespath python3-lib2to3 python3-libcomps python3-libdnf python3-looseversion python3-markupsafe
  python3-msgpack python3-numpy python3-olefile python3-packaging python3-pil python3-psutil python3-py python3-pycryptodome python3-qubesimgconverter python3-requests python3-rpm
  python3-tqdm python3-tz python3-unbound python3-yaml python3-zmq qubes-core-agent-dom0-updates qubes-core-agent-passwordless-root qubes-gpg-split qubes-img-converter
  qubes-input-proxy-sender qubes-mgmt-salt-vm-connector qubes-pdf-converter qubes-repo-templates qubes-usb-proxy rpm rpm-common rpm2cpio salt-common salt-ssh sqlite3 usbutils
```

Some user had issues with the switch to pipewire, myself included, and the only way to make sound work again was to install pulseaudio back.

**Comment by bi0shacker001** [comment]
If I do, for instance, "apt-get remove polkit" on a distribution, anything relying in polkit will also be removed. That is how package dependencies work.
Anything that was pulled in by those packages that is not otherwise in use would then be safe to remove. That is also how dependencies work. If it's not needed by anything that was explicitly installed, it would be safe to remove.
This process doesn't automatically remove those packages, users are free to flag them as needed if they want them. That doesn't change from any distribution. You are making a change from the defaults, and no matter which distribution you're in, that will come with headaches. I'm very confused as to why you wouldn't expect that from qubes.

**Comment by Minimalist73** [comment]
> If I do, for instance, "apt-get remove polkit" on a distribution, anything relying in polkit will also be removed. That is how package dependencies work.

This is about Qubes switching from pulseaudio to pipewire, the example don't fit the situation here.

The dependency to `qubes-vm-recommended` is here mainly because that's how the switch is done with the in-place [upgrade script](https://github.com/QubesOS/qubes-dist-upgrade/blob/main/scripts/upgrade-template-standalone.sh). Since `pipewire-qubes` is a part of `qubes-vm-recommanded`, it conflicts with pulseaudio and removes it, making it simpler to make the switch while upgrading. 

The only thing I'm bringing up here is that it's possible to easily switch from pipewire to pulseaudio in a Fedora template while user errors can be done on Debian while doing the same thing (if not protected, it removes important packages).

The package [`pulseaudio-qubes`](https://deb.qubes-os.org/r4.2/vm/pool/main/q/qubes-gui-agent/pulseaudio-qubes_4.2.9-1%2Bdeb12u1_amd64.deb) is still available in the Qubes Debian repository on 4.2 so I don't see why it couldn't be an option anymore while Fedora is? Based on some issues around pipewire (example: #8816) it should be possible, but the current implementation "blocks" it and I'm not aware of devs confirming pulseaudio support being deprecated on Debian.

**Comment by bi0shacker001** [comment]
> > If I do, for instance, "apt-get remove polkit" on a distribution, anything relying in polkit will also be removed. That is how package dependencies work.
> 
> This is about Qubes switching from pulseaudio to pipewire, the example don't fit the situation here.

It's an exact one-to-one. The issue you reported was that installing pulseaudio removes qubes-vm-recommended. That's how dependencies work

> The dependency to `qubes-vm-recommended` is here mainly because that's how the switch is done with the in-place [upgrade script](https://github.com/QubesOS/qubes-dist-upgrade/blob/main/scripts/upgrade-template-standalone.sh). Since `pipewire-qubes` is a part of `qubes-vm-recommended`, it conflicts with pulseaudio and removes it, making it simpler to make the switch while upgrading.
>
> The only thing I'm bringing up here is that it's possible to easily switch from pipewire to pulseaudio in a Fedora template while user errors can be done on Debian while doing the same thing (if not protected, it removes important packages).

In the fedora template, pulseaudio is still present, as well. I'd argue that you've pointed out an issue with fedora, rather than with the debian template, as pipewire and pulseaudio should conflict. It appears the rpms don't conflict. That's a potential issue, and should be reported upstream.

> The package [`pulseaudio-qubes`](https://deb.qubes-os.org/r4.2/vm/pool/main/q/qubes-gui-agent/pulseaudio-qubes_4.2.9-1%2Bdeb12u1_amd64.deb) is still available in the Qubes Debian repository on 4.2 so I don't see why it couldn't be an option anymore while Fedora is? Based on some issues around pipewire (example: #8816) it should be possible, but the current implementation "blocks" it and I'm not aware of devs confirming pulseaudio support being deprecated on Debian.
